### DISCUSSION TITLE
Use for iOS appstore links with Universal Links?

### DISCUSSION THREAD

**Discussion by pke** [fingerprintjs/fingerprintjs#810] [Q&A] [answered]
When users visit a link like `https://example.app?id=1` we would like to know `id` when starting the app for the first time after install. 

Android offers the _Install referrer API_ to easily append the `id` to the PlayStore link 
`https://play.google.com/store/apps/details?id=app.example&referrer=id%3D1` which allows the app to grab it (90 days after install) via a simple API inside the app. The request handler for `https://example.app?id=1` can easily create this redirect URL.

iOS however does not offer such feature. For iOS the `https://example.app?id=1` request handler would need to get a fingerprint of the users device/browser and save the `id` argument for later retrieval under the key, which is the fingerprint (`visitorId`?).
The iOS app then could open an invisible webview targetting `https://example.app?callback=app` indicating with `callback=app` to return the previously saved `id` in a redirect to the app. 

I know that iOS safari views no longer share cookies with the system safari app so I wonder if fingerprintjs would be of any help in this scenario? Give that the user on iOS could use any of Chrome, Safari or Firefox to open the initial `https://example.app?id=1` link I wonder if fingerprintjs can guarantee a stable fingerprint across all device browsers and from within the iOS app?

**edit**: I played around a bit and set up a HTML file that I opened from the safari browser in the simulator and added the react-native lib to my app. Bot using the same pro version public key. 
The app and safari get different visitorIds calculated. So that does not seem to work like intended.

**Comment by Valve** [comment, marked as answer]
@pke thanks for the question. 
We currently don't offer cross browser or app <=> browser identifiers.
If you generated a visitorID in webview and then later visited Safari, you'll get two different visitorIDs.
However we're working on a new product, called "browser linking", that will allow to link two different browser instances together. 
E.g. WebView ID: 12345
Safari ID: ABCDEF
Link: {ids: [12345, ABCDEF], confidence: 0.78}
If you're interested to discuss this linking product, please send me an email to Valentin at fingerprint dot com to get on a call.
Regarding the native APP ID vs browser IDs, these live completely separately and are not matching.
If you installed a native SDK and generated a visitorID there it will be different from a browser visitorID. 
There are no plans currently to link them together.

Some helpful links to other vendors/solutions:
https://firebase.google.com/docs/dynamic-links
https://help.adjust.com/en/article/deep-linking-ios-sdk

**Reply by pke** [reply]
Thanks for the answer. We have covered the Android side successfully by adding a referrer to the playstore URL we are redirecting to. 
Since iOS doesnt support this, it would certainly be interesting if browser linking could solve that problem for us, without resorting to adjust and the like. I will send you an email.
